### PR TITLE
[FLINK-8030] Instantiate JobMasterRestEndpoint in JobClusterEntrypoint

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

Launches the `JobMasterRestEndpoint` in the `JobClusterEntrypoint`.

PR is based on #4987.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by shuai-xu** [comment]
This pr is very helpful for us. Thank you. I have tried it on our cluster with yarn job mode，and it works well. My only suggestion is to change the default value of rest.address to 0.0.0.0 instead of localhost. Since if the rest server start at localhost, it can only be accessed from local machine. But if on 0.0.0.0, it can be accessed remotely.

**Comment by tillrohrmann** [comment]
Thanks for the review @shuai-xu. I'll change the default value of the rest address to `0.0.0.0`.

**Comment by tillrohrmann** [comment]
Actually we cannot change it to `0.0.0.0`, because the address is also used by the client to connect to the rest server. Therefore, it must be a valid target address.

File: flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+		this.metricFetcher = new MetricFetcher<>(
+			leaderRetriever,
+			metricQueryServiceRetriever,
+			executor,
+			restConfiguration.getTimeout());
+	}
+
+	@Override
+	protected List<Tuple2<RestHandlerSpecification, ChannelInboundHandler>> initializeHandlers(CompletableFuture<String> restAddressFuture) {
+		ArrayList<Tuple2<RestHandlerSpecification, ChannelInboundHandler>> handlers = new ArrayList<>(3);
```
Comment (GJL) [inline flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorEndpoint.java]: Initial size of 3 for the `ArrayList` isn't right anymore.

Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorEndpoint.java]: Good catch. Will correct it.

**Comment by tillrohrmann** [comment]
Thanks for the review @shuai-xu and @GJL. Merging this PR after Travis gives green light.

**Pull request merged**